src/main/java/org/folio/service/export/storage/AWSStorageServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package org.folio.service.export.storage;
+
+import static java.lang.System.getProperty;
+
+import java.lang.invoke.MethodHandles;
+import java.nio.file.Paths;
+
+import org.drools.core.util.StringUtils;
+import org.folio.rest.jaxrs.model.FileDefinition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import com.amazonaws.services.s3.transfer.MultipleFileUpload;
+import com.amazonaws.services.s3.transfer.TransferManager;
+
+/**
+ * Retrieve files that are stored in AWS S3
+ */
+@Service
+public class AWSStorageServiceImpl implements ExportStorageService {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  private static final String MARC_FILE_EXTENSION = "mrc";
+
+  @Autowired
+  private AmazonFactory amazonFactory;
+
+  /**
+   * Fetch the link to download a file for a given job by fileName
+   *
+   * @param jobId:   The job to which the files are associated
+   * @param tenantId
+   * @return A link using which the file can be downloaded
+   */
+  @Override
+  public String getFileDownloadLink(String jobId, String exportFileId, String tenantId) {
+    return StringUtils.EMPTY;
+  }
+
+  @Override
+  public void storeFile(FileDefinition fileDefinition, String tenantId) {
+    String parentFolder = String.format("%s/%s/%s.%s", tenantId, fileDefinition.getJobExecutionId(), fileDefinition.getId(), MARC_FILE_EXTENSION);
```

Comment:
shouldn't the extension be at the end of the file name? no at the parent folder level

Reply:
Also, don't have the fileDefinition Id in the path, while fetching the file , we'll just have access to jobExecution details

Reply:
so it has to be just tenantId/jobExecutionID/fileName.mrc

Reply:
The intention was to just copy all the files previously generated. There is a jira to address the missing extension. https://issues.folio.org/browse/MDEXP-72

Reply:
but here it's the parent folder string. If I'm not wrong the above STring will be "tenatid/jobExeId/fileDefId.mrc" which is incorrect isn;t it?

Reply:
fixed